File: tools/make_rois_pred.R
library(dplyr)
library(tidyr)

targets::tar_load(c(roi_avg, roi_avg_ukb))

roi_avg <- roi_avg |> dplyr::filter(n_parcels == 400)
roi_avg_ukb <- roi_avg_ukb |> dplyr::filter(n_parcels == 400)

rois <- bind_rows(roi_avg, roi_avg_ukb)
rois |> arrow::write_parquet("data-raw/rois0.parquet")

# gold <- rois |>
#   summarise(
#     d = abs(mean(value) / sd(value)),
#     .by = c(type, Task, label)
#   ) |>
#   mutate(r = rank(d, ties.method = "first"), .by = c(type, Task)) |>
#   filter(r <= 10)

# rois |>
#   inner_join(gold, by = join_by(type, Task, label)) |>
#   select(sub, type, task = Task, value, r) |>
#   pivot_wider(names_from = r, names_sort = TRUE) |>
#   mutate(confounds = FALSE, sub = as.integer(sub)) |>
#   arrow::write_parquet("data-raw/rois-for-prediction.parquet")

File: tools/make_rois_pred.py
import polars as pl

roi_avg = pl.read_parquet("../data-raw/rois0.parquet")

gold = (
    roi_avg.group_by("type", "Task", "label")
    .agg(d=pl.col("value").mean().abs() / pl.col("value").std())
    .with_columns(r=pl.col("d").rank(method="ordinal").over("type", "Task"))
    .filter(pl.col("r") < 11)
)

roi_avg.join(gold, how="inner", on=["type", "Task", "label"]).rename(
    {"Task": "task"}
).select("sub", "task", "type", "value", "r").sort("r").pivot(
    on="r", index=["sub", "task", "type"]
).with_columns(
    pl.col("sub").cast(pl.Int64),
    pl.col("type").cast(pl.Enum(categories=["MSMALL", "SURFACE", "VOL", "UKB"])),
    pl.col("task").cast(
        pl.Enum(
            categories=[
                "EMOTION",
                "GAMBLING",
                "LANGUAGE",
                "MOTOR",
                "RELATIONAL",
                "SOCIAL",
                "WM",
            ]
        )
    ),
    confounds=False,
).write_parquet("../data-raw/rois-for-prediction.parquet")
